## Case 1: Teaching Methods

### Q1

In [2]:
# group-wise mean/median/std
import numpy as np

group1 = [65, 70, 75, 80, 85, 78, 82, 79, 74, 68, 72, 77, 71, 84, 73]
group2 = [58, 62, 67, 65, 69, 70, 64, 60, 66, 68, 71, 59, 63, 67, 61]
group3 = [75, 78, 80, 85, 87, 82, 79, 88, 83, 81, 85, 84, 89, 86, 80]

for name, g in [("Group 1 (Traditional)", group1), ("Group 2 (Online)", group2), ("Group 3 (Blended)", group3)]:
    g = np.array(g)
    print(f"{name}: mean={g.mean():.2f}, median={np.median(g):.2f}, std={g.std(ddof=1):.2f}")

Group 1 (Traditional): mean=75.53, median=75.00, std=5.88
Group 2 (Online): mean=64.67, median=65.00, std=4.08
Group 3 (Blended): mean=82.80, median=83.00, std=3.99


### Q2

In [4]:
# one-way ANOVA using scipy
from scipy import stats

group1 = [65, 70, 75, 80, 85, 78, 82, 79, 74, 68, 72, 77, 71, 84, 73]
group2 = [58, 62, 67, 65, 69, 70, 64, 60, 66, 68, 71, 59, 63, 67, 61]
group3 = [75, 78, 80, 85, 87, 82, 79, 88, 83, 81, 85, 84, 89, 86, 80]

f_stat, p_value = stats.f_oneway(group1, group2, group3)
print("F-statistic:", f_stat)
print("p-value:", p_value)
if p_value < 0.05:
    print("Result: statistically significant difference between the groups (reject H0)")
else:
    print("Result: no statistically significant difference (fail to reject H0)")

F-statistic: 55.849985807550375
p-value: 1.4726026075567272e-12
Result: statistically significant difference between the groups (reject H0)


### Q3

In [6]:
# tukey hsd for pairwise comparisons since anova came back significant
import numpy as np
from statsmodels.stats.multicomp import pairwise_tukeyhsd

group1 = [65, 70, 75, 80, 85, 78, 82, 79, 74, 68, 72, 77, 71, 84, 73]
group2 = [58, 62, 67, 65, 69, 70, 64, 60, 66, 68, 71, 59, 63, 67, 61]
group3 = [75, 78, 80, 85, 87, 82, 79, 88, 83, 81, 85, 84, 89, 86, 80]

scores = group1 + group2 + group3
labels = ["Traditional"]*len(group1) + ["Online"]*len(group2) + ["Blended"]*len(group3)

tukey = pairwise_tukeyhsd(endog=scores, groups=labels, alpha=0.05)
print(tukey)

    Multiple Comparison of Means - Tukey HSD, FWER=0.05     
 group1    group2   meandiff p-adj   lower    upper   reject
------------------------------------------------------------
Blended      Online -18.1333    0.0  -22.329 -13.9377   True
Blended Traditional  -7.2667 0.0004 -11.4623   -3.071   True
 Online Traditional  10.8667    0.0    6.671  15.0623   True
------------------------------------------------------------


### Q4

In [8]:
# normality + equal variance checks
from scipy import stats

group1 = [65, 70, 75, 80, 85, 78, 82, 79, 74, 68, 72, 77, 71, 84, 73]
group2 = [58, 62, 67, 65, 69, 70, 64, 60, 66, 68, 71, 59, 63, 67, 61]
group3 = [75, 78, 80, 85, 87, 82, 79, 88, 83, 81, 85, 84, 89, 86, 80]

for name, g in [("Group 1", group1), ("Group 2", group2), ("Group 3", group3)]:
    stat, p = stats.shapiro(g)
    print(f"Shapiro-Wilk {name}: stat={stat:.4f}, p={p:.4f}")

levene_stat, levene_p = stats.levene(group1, group2, group3)
print(f"\nLevene's test: stat={levene_stat:.4f}, p={levene_p:.4f}")

Shapiro-Wilk Group 1: stat=0.9817, p=0.9800
Shapiro-Wilk Group 2: stat=0.9653, p=0.7826
Shapiro-Wilk Group 3: stat=0.9789, p=0.9613

Levene's test: stat=1.7028, p=0.1945


**Q5.** Blended learning comes out on top - highest mean (82.80) and it's significantly different from both other groups in the Tukey test. Traditional is second (75.53), and online learning is lowest (64.67). Normality and equal-variance assumptions both hold (all Shapiro p-values > 0.05, Levene's p = 0.19), so the ANOVA result can be trusted. if I had to recommend one method, it'd be blended learning.

## Case 2: Marketing Campaigns

### Q1

In [12]:
# region-wise mean/median/std
import numpy as np

north = [102, 110, 108, 104, 112, 109, 115, 114, 120, 118, 107, 113, 117, 121, 116, 111, 119, 122, 125, 126]
south = [95, 100, 98, 102, 97, 101, 104, 105, 108, 106, 99, 103, 107, 109, 110, 98, 102, 108, 105, 107]
west  = [110, 115, 120, 125, 123, 119, 121, 124, 127, 130, 118, 126, 129, 128, 131, 117, 122, 128, 132, 134]

for name, g in [("North", north), ("South", south), ("West", west)]:
    g = np.array(g)
    print(f"{name}: mean={g.mean():.2f}, median={np.median(g):.2f}, std={g.std(ddof=1):.2f}")

North: mean=114.45, median=114.50, std=6.68
South: mean=103.20, median=103.50, std=4.38
West: mean=123.95, median=124.50, std=6.23


### Q2

In [14]:
# one-way ANOVA across regions
from scipy import stats

north = [102, 110, 108, 104, 112, 109, 115, 114, 120, 118, 107, 113, 117, 121, 116, 111, 119, 122, 125, 126]
south = [95, 100, 98, 102, 97, 101, 104, 105, 108, 106, 99, 103, 107, 109, 110, 98, 102, 108, 105, 107]
west  = [110, 115, 120, 125, 123, 119, 121, 124, 127, 130, 118, 126, 129, 128, 131, 117, 122, 128, 132, 134]

f_stat, p_value = stats.f_oneway(north, south, west)
print("F-statistic:", f_stat)
print("p-value:", p_value)
if p_value < 0.05:
    print("Result: significant difference in sales between regions (reject H0)")
else:
    print("Result: no significant difference (fail to reject H0)")

F-statistic: 63.04200194761927
p-value: 3.6049308906779635e-15
Result: significant difference in sales between regions (reject H0)


### Q3

In [16]:
# tukey hsd - anova was significant so checking which regions differ
from statsmodels.stats.multicomp import pairwise_tukeyhsd

north = [102, 110, 108, 104, 112, 109, 115, 114, 120, 118, 107, 113, 117, 121, 116, 111, 119, 122, 125, 126]
south = [95, 100, 98, 102, 97, 101, 104, 105, 108, 106, 99, 103, 107, 109, 110, 98, 102, 108, 105, 107]
west  = [110, 115, 120, 125, 123, 119, 121, 124, 127, 130, 118, 126, 129, 128, 131, 117, 122, 128, 132, 134]

sales = north + south + west
labels = ["North"]*len(north) + ["South"]*len(south) + ["West"]*len(west)

tukey = pairwise_tukeyhsd(endog=sales, groups=labels, alpha=0.05)
print(tukey)

Multiple Comparison of Means - Tukey HSD, FWER=0.05 
group1 group2 meandiff p-adj  lower    upper  reject
----------------------------------------------------
 North  South   -11.25   0.0 -15.7022 -6.7978   True
 North   West      9.5   0.0   5.0478 13.9522   True
 South   West    20.75   0.0  16.2978 25.2022   True
----------------------------------------------------


### Q4

In [18]:
# normality + equal variance checks for the regions
from scipy import stats

north = [102, 110, 108, 104, 112, 109, 115, 114, 120, 118, 107, 113, 117, 121, 116, 111, 119, 122, 125, 126]
south = [95, 100, 98, 102, 97, 101, 104, 105, 108, 106, 99, 103, 107, 109, 110, 98, 102, 108, 105, 107]
west  = [110, 115, 120, 125, 123, 119, 121, 124, 127, 130, 118, 126, 129, 128, 131, 117, 122, 128, 132, 134]

for name, g in [("North", north), ("South", south), ("West", west)]:
    stat, p = stats.shapiro(g)
    print(f"Shapiro-Wilk {name}: stat={stat:.4f}, p={p:.4f}")

levene_stat, levene_p = stats.levene(north, south, west)
print(f"\nLevene's test: stat={levene_stat:.4f}, p={levene_p:.4f}")

Shapiro-Wilk North: stat=0.9855, p=0.9846
Shapiro-Wilk South: stat=0.9624, p=0.5929
Shapiro-Wilk West: stat=0.9799, p=0.9324

Levene's test: stat=1.6495, p=0.2012


### Q5

In [20]:
# one-sample t-test against the $110k target for each region
from scipy import stats

north = [102, 110, 108, 104, 112, 109, 115, 114, 120, 118, 107, 113, 117, 121, 116, 111, 119, 122, 125, 126]
south = [95, 100, 98, 102, 97, 101, 104, 105, 108, 106, 99, 103, 107, 109, 110, 98, 102, 108, 105, 107]
west  = [110, 115, 120, 125, 123, 119, 121, 124, 127, 130, 118, 126, 129, 128, 131, 117, 122, 128, 132, 134]

for name, g in [("North", north), ("South", south), ("West", west)]:
    t_stat, p_value = stats.ttest_1samp(g, 110)
    print(f"{name}: mean={sum(g)/len(g):.2f}, t={t_stat:.4f}, p={p_value:.4f}")

North: mean=114.45, t=2.9772, p=0.0077
South: mean=103.20, t=-6.9364, p=0.0000
West: mean=123.95, t=10.0172, p=0.0000


**Q6.** West region has the highest average sales (123.95k) and it's significantly higher than both North (114.45k) and South (103.20k) per Tukey. North does clear the $110k bar (t=2.98, p=0.008), West obviously does too (t=10.02, p<0.001) - but South falls significantly short of $110k (t=-6.94, p<0.001), so that campaign isn't meeting expectations. West's campaign looks like the most effective one, and South's needs a rethink since it's not just missing the target, it's missing it by a clear margin.